core/src/main/java/org/apache/gravitino/storage/relational/utils/POConverters.java

```diff
@@ -457,17 +457,35 @@ public static TablePO updateTablePOWithVersionAndSchemaId(
     }
 
     try {
-      return TablePO.builder()
-          .withTableId(oldTablePO.getTableId())
-          .withTableName(newTable.name())
-          .withMetalakeId(oldTablePO.getMetalakeId())
-          .withCatalogId(oldTablePO.getCatalogId())
-          .withSchemaId(newSchemaId)
-          .withAuditInfo(JsonUtils.anyFieldMapper().writeValueAsString(newTable.auditInfo()))
-          .withCurrentVersion(currentVersion)
-          .withLastVersion(lastVersion)
-          .withDeletedAt(DEFAULT_DELETED_AT)
-          .build();
+      TablePO.Builder builder =
+          TablePO.builder()
+              .withTableId(oldTablePO.getTableId())
+              .withTableName(newTable.name())
+              .withMetalakeId(oldTablePO.getMetalakeId())
+              .withCatalogId(oldTablePO.getCatalogId())
+              .withSchemaId(newSchemaId)
+              .withAuditInfo(JsonUtils.anyFieldMapper().writeValueAsString(newTable.auditInfo()))
+              .withCurrentVersion(currentVersion)
+              .withLastVersion(lastVersion)
+              .withDeletedAt(DEFAULT_DELETED_AT);
+
+      // Note: GenericTableEntity will be removed in the refactor PR, so here just keep the old
+      // logic to make the UT pass.
+      if (newTable instanceof GenericTableEntity genericTable) {
```

Comment:
This is a temporary change to make the CI pass, and #8922 will remove the class `GenericTableEntity` and use `TableEntity`.